recipes/catch2/2.x.x/conanfile.py

```diff
@@ -42,6 +41,7 @@ def config_options(self):
             del self.options.fPIC
 
     def configure(self):
+        self.package_type = "static-library" if self.options.with_main else "header-library"
```

Comment:
It can not be dynamically set AFAIK -- these should be different recipes to have this seperation

Reply:
```suggestion
```

Reply:
It can obviously since CI passes, and catch2 2.x can't be split in 2 different recipes.

Reply:
Sorry if I was unclear... but i think we are not on the same page here. Let me try again

CCI does not test the consumer graph combinations 😕 So I am not sure why that is relevant here.
The new graph was original designed using only options which is why that is the preferred way of doing it. (i.e omit the package type completely is supported)

I see you do not like the UX but the code you are adding is untested in the client to my knowledge and should not be done at this time. It's unclear what consequences it can have since the graph would change after being loaded and might have edge cases.